Azure Sentinel - 8296 - All the actions related to incidents (#1146)
rapid7/insightconnect-plugins: plugins/azure_sentinel/help.md

# Description

Azure Sentinel is Microsoft's' automated security service.

# Key Features

Identify key features of plugin.

# Requirements

* Example: Requires an API Key from the product
* Example: API must be enabled on the Settings page in the product's user interface

# Supported Product Versions

> _There are no supported product versions listed._

Author: Done!

```
```

##### Output

|Name|Type|Required|Description|
|----|----|--------|-----------|
|alerts|[]Alert|False|All the alerts assigned to the given incident.|

Example output:

```
```

#### List incidents

This action gets all the incidents.

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|api-version|string|2021-04-01|True|The API version to use for this operation.|None|None|
|filter|string|None|False|Filters the results, based on a Boolean condition. Optional.|None|None|
|orderBy|string|None|False|sorts the results|None|None|
|resourceGroupName|string|None|True|The name of the resource group within the user's subscription. The name is case insensitive.|None|None|
|subscriptionId|string|None|True|Azure subscription ID Regex pattern: ^[0-9A-Fa-f]{8}-([0-9A-Fa-f]{4}-){3}[0-9A-Fa-f]{12}$|None|None|
|top|integer|None|False|Return only n first results.|None|None|
|workspaceName|string|None|True|The name of the workspace|None|None|

Example input:

```
```

##### Output

|Name|Type|Required|Description|
|----|----|--------|-----------|
|incidents|[]Incident|False|list of incidents objects|

Example output:

```

> ```

Author: done

        },
        "relatedAnalyticRuleIds":[],
        "incidentUrl":"https://portal.azure.com/#asset/Microsoft_Azure_Security_Insights/Incident/subscriptions/eeeee-eeee-eee-eeee-df965e27aefe/resourceGroups/integrationLab/providers/Microsoft.OperationalInsights/workspaces/sentinel/providers/Microsoft.SecurityInsights/Incidents/1407186457"
      }
    }
  ]
}
```
#### Create or Update Incident

This action this action creates or updates an incident.

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|etag|string|None|False|Etag of the azure resource|None|0300bf09-0000-0000-0000-5c37296e0000|
|incidentId|string|None|True|Incident ID|None|incident-14071867|
|properties|IncidentProperties|None|True|Incident properties object|None|{'status': 'Closed'}|
|resourceGroupName|string|None|True|The name of the resource group within the user's subscription. The name is case-insensitive|None|resourcegrup1|
|subscriptionId|string|None|True|Azure subscription ID|None|aaaef455-a780-44ca-9e51-aaafffeeea3a|
|workspaceName|string|None|True|The name of the workspace|None|workspace23|

Example input:

```

> ```

Reviewer: Empty one 😄

Author: This pesky help.md

|Azure resource type|string|False|Azure resource type|

#### SystemData

|Name|Type|Required|Description|
|----|----|--------|-----------|
|The timestamp of resource creation (UTC).|date|False|The timestamp of resource creation (UTC).|
|The identity that created the resource.|string|False|The identity that created the resource.|
|The type of identity that created the resource.|CreatedByType|False|The type of identity that created the resource.|
|The timestamp of resource last modification (UTC)|date|False|The timestamp of resource last modification (UTC)|
|The identity that last modified the resource.|string|False|The identity that last modified the resource.|
|The type of identity that last modified the resource.|CreatedByType|False|The type of identity that last modified the resource.|

#### UserInfo

|Name|Type|Required|Description|
|----|----|--------|-----------|
|Email|string|False|The email of the user.|
|Name|string|False|The name of the user.|
|Object Identification|string|False|The object id of the user.|

## Troubleshooting

_This plugin does not contain any troubleshooting information._

# Version History

> * 1.0.0 - Initial plugin

Reviewer: Please add all included actions in the open brackets
```suggestion
* 1.0.0 - Initial plugin (Actions: Action 1, Action2 ...)
```

# Description

Azure Sentinel is Microsoft's' automated security service.

# Key Features

* Creating and updating incidents
* Deleting incidents

> * Retriveing incident's details

Reviewer 2: *an incident's

* Listing incidents for a given workspace
* Listing bookmarks for a given incident
* Listing alerts for a given incident

# Requirements

> * Set of credentials with necessary permissions to monitor and modify Sentinel incidents

Reviewer 2: "Set of Azure credentials" (just try to be specific)

# Supported Product Versions

* 2021-04-01

# Documentation

## Setup

The connection configuration accepts the following parameters:

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|client_id|string|None|True|The application ID that the application registration portal assigned to your app|None|c163eff0-d1a1-4618-ee2a-453534f43cee|
|client_secret|credential_secret_key|None|True|The application secret that you generated for your app in the app registration portal|None|ef50c6bx9umaik9agvoxtoqec2fg9f0y|

> |tenant_id|string|None|True|This is active directory ID|None|5ceea899-ae8c-4ff1-fffe-353646eeeff0|

Reviewer 2: "This is active directory ID" → "The Active Directory ID" [for what? from where?]

Example input:

```
{
  "client_id": "c163eff0-d1a1-4618-ee2a-453534f43cee",
  "client_secret": "ef50c6bx9umaik9agvoxtoqec2fg9f0y",
  "tenant_id": "5ceea899-ae8c-4ff1-fffe-353646eeeff0"
}
```

## Technical Details

### Actions

#### List Entities

> This action is used to get all incidents entities.

Reviewer 2: ```suggestion
This action is used to get all entities from a given incident.
```

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|incidentId|string|None|True|Incident ID|None|incident123|
|resourceGroupName|string|None|True|The name of the resource group within the user's subscription|None|resourcegrup12|
|subscriptionId|string|None|True|Azure subscription ID|None|0caafeeb-aaa0-44ca-ffe1-aaaaeeeffffe|
|workspaceName|string|None|True|The name of the workspace|None|workspace23|

Example input:

```
{
  "incidentId": "incident123",
  "resourceGroupName": "resourcegrup12",
  "subscriptionId": "0caafeeb-aaa0-44ca-ffe1-aaaaeeeffffe",
  "workspaceName": "workspace23"
}
```

##### Output

|Name|Type|Required|Description|
|----|----|--------|-----------|
|entities|[]Entity|False|All the entities assigned to the given incident|

Example output:

```
{
  "entities": []
}
```

#### List Incidents

This action is used to list all the incidents matching specified criteria.

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|orderBy|string|None|False|Field to sort results by|None|properties/createdTimeUtc desc|

> |resourceGroupName|string|None|True|The name of the resource group within the user's subscription. The name is case-insensitive|None|resourcegrup1|

Reviewer 2: ```suggestion
|resourceGroupName|string|None|True|The name of the resource group within the user's subscription. Case-insensitive.|None|resourcegrup1|
```

      "type":"Microsoft.SecurityInsights/Incidents",
      "properties":{
        "title":"Test incident",
        "description":"This is a demo incident",
        "severity":"Low",
        "status":"Closed",
        "classification":"FalsePositive",
        "classificationReason":"IncorrectAlertLogic",
        "classificationComment":"Not a malicious activity",
        "labels":[],
        "firstActivityTimeUtc":"2019-01-01T13:00:30Z",
        "lastActivityTimeUtc":"2019-01-01T13:05:30Z",
        "lastModifiedTimeUtc":"2022-02-23T16:52:50.1945053Z",
        "createdTimeUtc":"2022-02-23T16:52:50.1945053Z",
        "incidentNumber":3,
        "additionalData":{
          "alertsCount":0,
          "bookmarksCount":0,
          "commentsCount":0,
          "alertProductNames":[]
        },
        "relatedAnalyticRuleIds":[],
        "incidentUrl":"https://portal.azure.com/#asset/Microsoft_Azure_Security_Insights/Incident/subscriptions/eeeee-eeee-eee-eeee-df965e27aefe/resourceGroups/integrationLab/providers/Microsoft.OperationalInsights/workspaces/sentinel/providers/Microsoft.SecurityInsights/Incidents/1407186457"
      }
    }
  ]
}
```
#### Create or Update Incident

> This action this action creates or updates an incident.

Reviewer 2: ```suggestion
Create a new incident or update an existing incident.
```

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|

> |etag|string|None|False|Etag of the azure resource|None|0300bf09-0000-0000-0000-5c37296e0000|

Reviewer 2: Thought: where can the user find the etag?

|incidentId|string|None|True|Incident ID|None|incident-14071867|

> |properties|IncidentProperties|None|True|Incident properties object|None|{'status': 'Closed'}|

Reviewer 2: A JSON object of keys and values of properties for the incident

  "properties":{
    "title":"Incident At Work",
    "description":"This is a demo incident",
    "severity":"High",
    "status":"Closed",
    "classification":"FalsePositive",
    "classificationReason":"IncorrectAlertLogic",
    "classificationComment":"Not a malicious activity",
    "owner":{
      "objectId":"2046feea-040d-4a46-9e2b-91c2941bfa70"
    },
    "labels":[
      
    ],
    "firstActivityTimeUtc":"2019-01-01T13:00:30Z",
    "lastActivityTimeUtc":"2019-01-01T13:05:30Z",
    "lastModifiedTimeUtc":"2022-03-08T13:49:39.1064183Z",
    "createdTimeUtc":"2022-02-09T13:05:21.7201975Z",
    "incidentNumber":2,
    "additionalData":{
      "alertsCount":0,
      "bookmarksCount":0,
      "commentsCount":0,
      "alertProductNames":[
        
      ],
      "tactics":[
        
      ]
    },
    "relatedAnalyticRuleIds":[
      
    ],
    "incidentUrl":"https://portal.azure.com/#asset/Microsoft_Azure_Security_Insights/Incident/subscriptions/aaaef455-a780-44ca-9e51-aaafffeeea3a/resourceGroups/integrationLab/providers/Microsoft.OperationalInsights/workspaces/sentinel/providers/Microsoft.SecurityInsights/Incidents/14071867"
  }
}
```

#### Delete Incident

> This action is used to get all incidents from a given workspace and resource group.

Reviewer 2: Wrong description?

```suggestion
Delete an incident from the system.
```

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|

> |incidentId|string|None|True|Incident ID|None|incident123|

Reviewer 2: ```suggestion
|incidentId|string|None|True|ID of the incident to delete|None|incident123|
```

|resourceGroupName|string|None|True|The name of the resource group within the user's subscription. The name is case-insensitive|None|resourcegrup1|
|subscriptionId|string|None|True|Azure subscription ID|None|0caafeeb-aaa0-44ca-ffe1-aaaaeeeffffe|
|workspaceName|string|None|True|The name of the workspace|None|workspace23|

Example input:

```
{
  "incidentId": "incident123",
  "resourceGroupName": "resourcegrup1",
  "subscriptionId": "0caafeeb-aaa0-44ca-ffe1-aaaaeeeffffe",
  "workspaceName": "workspace23"
}
```

##### Output

|Name|Type|Required|Description|
|----|----|--------|-----------|
|status|integer|False|Deletion status, 200 - ok, 204 - no content|

Example output:

```
{
  "status": 204
}
```

#### List Alerts

> This action is used to get all incidents alerts.

Reviewer 2: ```suggestion
This action is used to get all alerts for a given incident.
```

      "name":"baa8a239-6fde-4ab7-a093-d09f7b75c58c",
      "type":"Microsoft.SecurityInsights/Entities",
      "kind":"SecurityAlert",
      "properties":{
        "systemAlertId":"baa8a239-6fde-4ab7-a093-d09f7b75c58c",
        "tactics":[
          
        ],
        "alertDisplayName":"myAlert",
        "confidenceLevel":"Unknown",
        "severity":"Low",
        "vendorName":"Microsoft",
        "productName":"Azure Security Center",
        "alertType":"myAlert",
        "processingEndTime":"2020-07-20T18:21:53.6158361Z",
        "status":"New",
        "endTimeUtc":"2020-07-20T18:21:53.6158361Z",
        "startTimeUtc":"2020-07-20T18:21:53.6158361Z",
        "timeGenerated":"2020-07-20T18:21:53.6158361Z",
        "resourceIdentifiers":[
          {
            "type":"LogAnalytics",
            "workspaceId":"c8c99641-985d-4e4e-8e91-fb3466cd0e5b",
            "subscriptionId":"bd794837-4d29-4647-9105-6339bfdb4e6a",
            "resourceGroup":"myRG"
          }
        ],
        "additionalData":{
          "AlertMessageEnqueueTime":"2020-07-20T18:21:57.304Z"
        },
        "friendlyName":"myAlert"
      }
    }
  ]
}
```

#### Get Incident

> This action is used to get all incidents from a given workspace and resource group.

Reviewer 2: ```suggestion
This action is used to get details for one specific incident.
```

  "properties":{
    "title":"Incident At Work",
    "description":"This is a demo incident",
    "severity":"High",
    "status":"Closed",
    "classification":"FalsePositive",
    "classificationReason":"IncorrectAlertLogic",
    "classificationComment":"Not a malicious activity",
    "owner":{
      "objectId":"2046feea-040d-4a46-9e2b-91c2941bfa70"
    },
    "labels":[
      
    ],
    "firstActivityTimeUtc":"2019-01-01T13:00:30Z",
    "lastActivityTimeUtc":"2019-01-01T13:05:30Z",
    "lastModifiedTimeUtc":"2022-03-08T13:49:39.1064183Z",
    "createdTimeUtc":"2022-02-09T13:05:21.7201975Z",
    "incidentNumber":2,
    "additionalData":{
      "alertsCount":0,
      "bookmarksCount":0,
      "commentsCount":0,
      "alertProductNames":[
        
      ],
      "tactics":[
        
      ]
    },
    "relatedAnalyticRuleIds":[
      
    ],
    "incidentUrl":"https://portal.azure.com/#asset/Microsoft_Azure_Security_Insights/Incident/subscriptions/aaaef455-a780-44ca-9e51-aaafffeeea3a/resourceGroups/integrationLab/providers/Microsoft.OperationalInsights/workspaces/sentinel/providers/Microsoft.SecurityInsights/Incidents/14071867"
  }
}
```

#### List Bookmarks

> This action is used to get all incidents bookmarks.

Reviewer 2: ```suggestion
This action is used to get all bookmarks for a given incident.
```

|The tactics associated with incident|[]string|False|The tactics associated with incident|

#### IncidentLabel

|Name|Type|Required|Description|
|----|----|--------|-----------|
|The name of the label|string|False|The name of the label|
|The type of label|string|False|The type of label.|

#### IncidentOwnerInfo

|Name|Type|Required|Description|
|----|----|--------|-----------|
|The name of the user the incident is assigned to.|string|False|The name of the user the incident is assigned to.|
|The email of the user the incident is assigned to.|string|False|The email of the user the incident is assigned to.|
|The object id of the user the incident is assigned to.|string|False|The object id of the user the incident is assigned to.|
|The user principal name of the user the incident is assigned to.|string|False|The user principal name of the user the incident is assigned to.|

#### IncidentProperties

|Name|Type|Required|Description|
|----|----|--------|-----------|
|Additional data on the incident|IncidentAdditionalData|False|Additional data on the incident|
|The reason the incident was closed|string|False|The reason the incident was closed|

> |Describes the reason the incident was closed|string|False|Describes the reason the incident was closed|

Reviewer 2: Are these really the property names?

Author: It seems there is a bug in the generation of `help.md`

Example input:

```
{
  "client_id": "c163eff0-d1a1-4618-ee2a-453534f43cee",
  "client_secret": "ef50c6bx9umaik9agvoxtoqec2fg9f0y",
  "tenant_id": "5ceea899-ae8c-4ff1-fffe-353646eeeff0"
}
```

## Technical Details

### Actions

#### List Entities

This action is used to get all incidents entities.

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|incidentId|string|None|True|Incident ID|None|incident123|
|resourceGroupName|string|None|True|The name of the resource group within the user's subscription|None|resourcegrup12|
|subscriptionId|string|None|True|Azure subscription ID|None|0caafeeb-aaa0-44ca-ffe1-aaaaeeeffffe|
|workspaceName|string|None|True|The name of the workspace|None|workspace23|

Example input:

```
{
  "incidentId": "incident123",
  "resourceGroupName": "resourcegrup12",
  "subscriptionId": "0caafeeb-aaa0-44ca-ffe1-aaaaeeeffffe",
  "workspaceName": "workspace23"
}
```

##### Output

|Name|Type|Required|Description|
|----|----|--------|-----------|
|entities|[]Entity|False|All the entities assigned to the given incident|

Example output:

```

> {

Reviewer 3: Updated

      "type":"Microsoft.SecurityInsights/Incidents",
      "properties":{
        "title":"Test incident",
        "description":"This is a demo incident",
        "severity":"Low",
        "status":"Closed",
        "classification":"FalsePositive",
        "classificationReason":"IncorrectAlertLogic",
        "classificationComment":"Not a malicious activity",
        "labels":[],
        "firstActivityTimeUtc":"2019-01-01T13:00:30Z",
        "lastActivityTimeUtc":"2019-01-01T13:05:30Z",
        "lastModifiedTimeUtc":"2022-02-23T16:52:50.1945053Z",
        "createdTimeUtc":"2022-02-23T16:52:50.1945053Z",
        "incidentNumber":3,
        "additionalData":{
          "alertsCount":0,
          "bookmarksCount":0,
          "commentsCount":0,
          "alertProductNames":[]
        },
        "relatedAnalyticRuleIds":[],
        "incidentUrl":"https://portal.azure.com/#asset/Microsoft_Azure_Security_Insights/Incident/subscriptions/eeeee-eeee-eee-eeee-df965e27aefe/resourceGroups/integrationLab/providers/Microsoft.OperationalInsights/workspaces/sentinel/providers/Microsoft.SecurityInsights/Incidents/1407186457"
      }
    }
  ]
}
```
#### Create or Update Incident

> This action this action creates or updates an incident.

Reviewer 3: Updated

|Severity|string|True|Incidents severity|
|Status|string|True|Incidents status|
|System Data|SystemData|False|Azure Resource Manager metadata containing createdBy and modifiedBy information|
|The title of the incident|string|False|The title of the incident|
|Type|string|False|Azure resource type|

#### SystemData

|Name|Type|Required|Description|
|----|----|--------|-----------|
|Created At|date|False|The timestamp of resource creation (UTC)|
|Created By|string|False|The identity that created the resource|
|Created By Type|CreatedByType|False|The type of identity that created the resource|
|Last Modified At|date|False|The timestamp of resource last modification (UTC)|
|Last Modified By|string|False|The identity that last modified the resource|
|Last Modified By Type|CreatedByType|False|The type of identity that last modified the resource|

#### UserInfo

|Name|Type|Required|Description|
|----|----|--------|-----------|
|Email|string|False|The email of the user|
|Name|string|False|The name of the user|
|Object Identification|string|False|The object ID of the user|

## Troubleshooting

_This plugin does not contain any troubleshooting information._

# Version History

> * 1.0.0 - Initial plugin Initial plugin (Actions: Create or Update Incident, Delete Incident, List Incidents, Get Incident, List Alerts, List Bookmarks, List Entities)

Reviewer 3: Updated

        "additionalData":{
          "alertsCount":0,
          "bookmarksCount":0,
          "commentsCount":0,
          "alertProductNames":[]
        },
        "relatedAnalyticRuleIds":[],
        "incidentUrl":"https://portal.azure.com/#asset/Microsoft_Azure_Security_Insights/Incident/subscriptions/eeeee-eeee-eee-eeee-df965e27aefe/resourceGroups/integrationLab/providers/Microsoft.OperationalInsights/workspaces/sentinel/providers/Microsoft.SecurityInsights/Incidents/1407186457"
      }
    }
  ]
}
```
#### Create or Update Incident

This action creates or updates an incident.

##### Input

|Name|Type|Default|Required|Description|Enum|Example|
|----|----|-------|--------|-----------|----|-------|
|incidentId|string|None|True|Incident ID|None|incident-14071867|
|properties|IncidentProperties|None|True|Incident properties object|None|{'status': 'Closed'}|
|resourceGroupName|string|None|True|The name of the resource group within the user's subscription|None|resourcegrup1|
|subscriptionId|string|None|True|Azure subscription ID|None|aaaef455-a780-44ca-9e51-aaafffeeea3a|
|workspaceName|string|None|True|The name of the workspace|None|workspace23|

Example input:

> ```

Reviewer 4: Kindly add the input. Seems like the properties in the input is a complex object and seems like help.md did not render correctly. Can you please check.
